app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VotesController < ApplicationController
+  before_action :find_micropost
+  
+  def create
+
+    if already_voted?
+      flash[:warning] = "You can't vote more than once"
+    else
+      # I know it is not safely because someone can hack database by coding in form (or somehow).
+      # Does validation in model (1..5) solve this problem ?
+     @vote = @micropost.votes.create(grades: params[:grades], user_id: current_user.id) 
```

Comment:
If you are not using your vote object in a view, there is no reason to declare it as a instance variable, you could go with a local variable like `vote = @micropost.votes.create(grades: params[:grades], user_id: current_user.id)`